fxrecorder/src/lib/analysis.rs

```diff
@@ -0,0 +1,406 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::ffi::OsStr;
+use std::fs::{create_dir_all, read_dir, File};
+use std::io::{self, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+
+use image::{GenericImageView, ImageError, Rgb};
+use serde::{Deserialize, Serialize};
+use slog::{error, info, warn};
+use thiserror::Error;
+
+use crate::ffmpeg::{run_ffmpeg, FfmpegError};
+
+#[derive(Debug, Error)]
+#[error("Could not crop video: {}", .0)]
+pub struct CropVideoError(#[source] pub FfmpegError);
+
+/// Crop the video.
+pub fn crop_video(
+    log: slog::Logger,
+    video_path: &Path,
+    target_directory: &Path,
+) -> Result<PathBuf, CropVideoError> {
+    let output_path = target_directory.join("cropped.mp4");
+
+    let args = vec![
+        OsStr::new("-i"),
+        video_path.as_os_str(),
+        // The crop filter is used to remove the task bar, which is 40px high
+        // plus 1px for blurring due to compression.
+        OsStr::new("-vf"),
+        OsStr::new("crop=in_w:in_h-41:0:0"),
```

Comment:
Nit: may be helpful to pull up this cropping amount into a constant?

Reply:
Pulling out the entire string here doesn't make sense since its already commented to its purpose and is only used in one place.

Pulling out the literal 41 also doesn't make sense since it is again only used in one place and it is documented. Additionally, we would have to use format!() to recreate the string, which allocates, or pull in another crate to have a compile time format.

Reply:
However i failed to save after reverting pulling out the constant so i guess it stays heh